software/main/power.c

```diff
@@ -1,17 +1,20 @@
 #include "power.h"
 
-struct power_handle {
-  uint8_t reserved;
+struct power_handle
+{
+    uint8_t reserved;
 } handle;
 
-struct power_handle *power_init(struct power_config *config) {
-  (void)config;
+struct power_handle *power_init(struct power_config *config)
+{
+    (void)config;
 
-  return &handle;
+    return &handle;
 }
 
-double power_get_price(struct power_handle *handle) {
-  (void)handle;
+double power_get_price(struct power_handle *handle)
+{
+    (void)handle;
 
-  return 0.0;
+    return 0.0;
```

Comment:
How do you distinguish between when the value returned is actual $0.0 or an error condition (NaN)?

Reply:
The error condition for this function is returning `DBL_MAX`. I've assumed that the electricity price won't get that high :crossed_fingers:. This is just some random stub value 0.0 in this cause until the actual function is implemented.

_From power.h_

```c
@return Current electricity price in dollars on success, @c DBL_MAX on error.
```